starknet/test/stark-sig-auth.test.ts

```diff
@@ -0,0 +1,256 @@
+      author: address0,
+      proposalId: { low: '0x1', high: '0x0' },
+      executionStrategy: {
+        address: '0x0000000000000000000000000000000000005678',
+        params: ['0x5', '0x6', '0x7', '0x8'],
+      },
+      metadataURI: ['0x1', '0x2', '0x3', '0x4'],
+      salt: '0x1',
+    };
+    const updateProposalData: typedData.TypedData = {
+      types: updateProposalTypes,
+      primaryType: 'UpdateProposal',
+      domain: domain,
+      message: updateProposalMsg as any,
+    };
+
+    const updateProposalSig = (await account0.signMessage(updateProposalData)) as any;
+
+    const updateProposalCalldata: StarknetSigUpdateProposalCalldata = {
+      signature: [updateProposalSig.r, updateProposalSig.s],
+      ...updateProposalMsg,
+      accountType: account0Type,
+    };
+
+    await account0.execute({
+      contractAddress: starkSigAuthAddress,
+      entrypoint: 'authenticate_update_proposal',
+      calldata: CallData.compile(updateProposalCalldata as any),
+    });
+
+    {
+      // Random Tx just to advance the block number on the devnet so the voting period begins.
+
+      await account0.declareAndDeploy({
+        contract: json.parse(
+          fs
+            .readFileSync('starknet/target/dev/sx_StarkSigAuthenticator.sierra.json')
+            .toString('ascii'),
+        ),
+        casm: json.parse(
+          fs
+            .readFileSync('starknet/target/dev/sx_StarkSigAuthenticator.casm.json')
+            .toString('ascii'),
+        ),
+        constructorCalldata: CallData.compile({ name: 'sx-sn', version: '0.1.0' }),
+      });
+
+      await account0.declareAndDeploy({
+        contract: json.parse(
+          fs
+            .readFileSync('starknet/target/dev/sx_StarkSigAuthenticator.sierra.json')
+            .toString('ascii'),
+        ),
+        casm: json.parse(
+          fs
+            .readFileSync('starknet/target/dev/sx_StarkSigAuthenticator.casm.json')
+            .toString('ascii'),
+        ),
+        constructorCalldata: CallData.compile({ name: 'sx-sn', version: '0.1.0' }),
+      });
+    }
+
+    // VOTE
+
+    const voteMsg: Vote = {
+      space: spaceAddress,
+      voter: address0,
+      proposalId: { low: '0x1', high: '0x0' },
+      choice: '0x1',
+      userVotingStrategies: [{ index: '0x0', params: ['0x1', '0x2', '0x3', '0x4'] }],
+      metadataURI: ['0x1', '0x2', '0x3', '0x4'],
+    };
+
+    const voteData: typedData.TypedData = {
+      types: voteTypes,
+      primaryType: 'Vote',
+      domain: domain,
+      message: voteMsg as any,
+    };
+
+    const voteSig = (await account0.signMessage(voteData)) as any;
+
+    const voteCalldata: StarknetSigVoteCalldata = {
+      signature: [voteSig.r, voteSig.s],
+      ...voteMsg,
+      accountType: account0Type,
+    };
+
+    await account0.execute({
+      contractAddress: starkSigAuthAddress,
+      entrypoint: 'authenticate_vote',
+      calldata: CallData.compile(voteCalldata as any),
+    });
+  }, 1000000);
+});
```

Comment:
would be nice to have failing tests by providing wrong signatures / wrong signer

Reply:
yeah i have another pr with stark sig tests coming soon